File: framework/languages/java/e2e-kotlin/SpotService/src/main/java/systems/zlink/e2e/kotlin/spotservice/StateTimerHandler.java
package systems.zlink.e2e.kotlin.spotservice;

import systems.zlink.framework.spots.ZLinkSpotTimerHandler;
import systems.zlink.framework.spots.ZLinkTimerTick;

public final class StateTimerHandler implements ZLinkSpotTimerHandler<UserSpot> {
    @Override
    public void handle(
        UserSpot spot,
        ZLinkTimerTick tick) {
        spot.timerTick(tick.deliveryIndex());
    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Server/Play/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/play/handlers/StateTimerHandler.kt
package systems.zlink.e2e.kotlin.spotservice.play.handlers

import systems.zlink.e2e.kotlin.spotservice.Contracts
import systems.zlink.e2e.kotlin.spotservice.ScenarioState
import systems.zlink.e2e.kotlin.spotservice.play.spots.*
import systems.zlink.framework.spots.ZLinkSpotTimerHandler
import systems.zlink.framework.spots.ZLinkTimerTick

class StateTimerHandler : ZLinkSpotTimerHandler<UserSpot> {
    override fun handle(spot: UserSpot, tick: ZLinkTimerTick) {
        spot.timerTick(tick.deliveryIndex())
    }
}
